### PR TITLE
FIltering checks the tagsearch input against a tag matcher while stil…

### PR DESCRIPTION
**Pull request by Stormgun**
…l maintaining its original functionality, for finding partial matches.

The media objects that are returned by the tag matcher will be displayed or highlighted as appropriate

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "test": "mocha"
   },
   "engines": {
-    "node": "4.8.0"
+    "node": "6.3.1"
```
Comment (aaronpcz) [inline package.json]: We have made a change like to master I believe, we might want to resolve using merging ?

Comment (Stormgun) [reply package.json]: I'll take the change from master when I merge the master and dev .

**Comment by Stormgun** [comment]
reverted the package.json change

File: src/js/components/scene-editor/media-object-list.jsx [hunk 2]
```diff
@@ -89,21 +90,45 @@ var MediaObjectList = React.createClass({
...
         var items = null;
+        var self =this;
+        var mediaArray = null;
         try {
             if (this.props.scene && this.props.scene.scene && this.props.scene.scene.length !== 0) {
+
+                //AP: attaching tag matcher to the media objects array and plugging in the tagSearch value.
+                var match = _(self.props.scene.scene)
+                    .filter(function(mo) {
+                        return  new TagMatcher("(" + self.state.tagSearch + ")").match(mo.tags);
```
Comment (aaronpcz) [inline src/js/components/scene-editor/media-object-list.jsx]: Well worth pulling the tag matcher as a variable outside of the foreach, given it's the same each time.

Small performance improvement.

**Comment by aaronpcz** [comment]
@Stormgun I did a nice refactor job late last night - check it out and we can get it released so authors have a better tool for writing themes.

**Pull request merged**